actions/kubernetes_helpers.go

```diff
@@ -15,31 +16,45 @@ import (
 )
 
 func patchNode(ctx context.Context, clientset kubernetes.Interface, node *v1.Node, changeFn func(*v1.Node) error) error {
-	oldData, err := json.Marshal(node)
+	return patchObject[v1.Node](
+		ctx, node, changeFn,
+		func(ctx context.Context, patchType apitypes.PatchType, data []byte, options metav1.PatchOptions) error {
+			_, err := clientset.CoreV1().Nodes().Patch(ctx, node.Name, patchType, data, options)
+			return err
+		},
+	)
+}
+
+func patchObject[T any](
+	ctx context.Context,
+	object *T,
+	modify func(*T) error,
```

Comment:
do we need `modify` as part of generic method or can we pass already modified object straight to patch?

Reply:
would need to deep copy the object, the implemation is 
1. marshal current 
2. modify current
3. marshal modified
4. get diff